### PR TITLE
Story/11692 update logic to support rapid PCR tests

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Update business logic to support Rapid PCR Tests.

## Link to Jira
(https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11356)

## Screenshots

Screenshot 2022-02-08 at 14 46 51

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 1]
```diff
@@ -85,10 +84,69 @@ enum Route: Equatable {
...
+				return nil
+			}
+			// extract payload
+			guard let testInformation = RapidTestQRCodeInformation(payload: payloadUrl) else {
+				self = .rapidPCR( .failure(.invalidTestCode(.invalidPayload)))
+				Log.error("RapidPCR test data is nil, either timeStamp is -ve or the hash is invalid", log: .qrCode)
+				return
+			}
+			guard testInformation.hash.range(of: #"^[0-9A-Fa-f]{64}$"#, options: .regularExpression) != nil  else {
+				self = .rapidPCR( .failure(.invalidTestCode(.invalidHash)))
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:101]:
```suggestion
				self = .rapidPCR(.failure(.invalidTestCode(.invalidHash)))
```

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 2]
```diff
@@ -85,10 +84,69 @@ enum Route: Equatable {
...
+		case "p.coronawarn.app":
+			guard let payloadUrl = components?.fragment,
+				  let candidate = components?.query,
+				  candidate.count == 3 else {
+				Log.error("RapidPCR test QRCode URL is invalid", log: .qrCode)
+				return nil
+			}
+			// extract payload
+			guard let testInformation = RapidTestQRCodeInformation(payload: payloadUrl) else {
+				self = .rapidPCR( .failure(.invalidTestCode(.invalidPayload)))
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:96]:
```suggestion
				self = .rapidPCR(.failure(.invalidTestCode(.invalidPayload)))
```

File: src/xcode/ENA/ENA/Source/Models/Submisstion/RapidTestQRCodeInformation.swift [hunk 3]
```diff
@@ -4,7 +4,9 @@
 
 import Foundation
 
-struct AntigenTestQRCodeInformation: Codable, Equatable {
+// The Struct is used to store QRCode information for Rapid antigen tests and Rapid PCR Tests
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Models/Submisstion/RapidTestQRCodeInformation.swift:8]:
```suggestion
```

File: src/xcode/ENA/ENA/Source/Models/Submisstion/RapidTestQRCodeInformation.swift [hunk 4]
```diff
@@ -4,7 +4,9 @@
...
+struct RapidTestQRCodeInformation: Codable, Equatable {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Models/Submisstion/RapidTestQRCodeInformation.swift:9]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/Model/Parser/QRCodeParser.swift [hunk 5]
```diff
@@ -44,8 +44,9 @@ class QRCodeParser: QRCodeParsable {
 
 		let antigenTestPrefix = "https://s.coronawarn.app"
 		let pcrTestPrefix = "https://localhost"
+		let rapidPCRTestPrefix = "https://p.coronawarn.app"
```
[the remark this answers is not in the document]
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/Model/Parser/QRCodeParser.swift:47]: they are only used inside this function so far

**Review by naveeddotio — APPROVED** [review]

**Comment by KathrinLuecke** [comment]
Hi, no UI strings have been changed by this PR - why should we from user assistance review it? 
Is this some automatic assignment?

**Pull request merged**